app/src/main/java/com/spendster/presentation/addExpenses/chooseCategory/RecyclerViewAdapter.java

```diff
@@ -0,0 +1,72 @@
+package com.spendster.presentation.addExpenses.chooseCategory;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.CardView;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.spendster.R;
+import com.spendster.data.entity.Category;
+
+import java.util.List;
+
+public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.CategoryViewHolder> {
+
+    private Context context;
+    private List<Category> categories;
+    private OnItemClickListener onItemClickListener;
+
+    public RecyclerViewAdapter(Context context, List<Category> categories, OnItemClickListener onItemClickListener) {
+        this.context = context;
+        this.categories = categories;
+        this.onItemClickListener = onItemClickListener;
+    }
+
+    @NonNull
+    @Override
+    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        View view = LayoutInflater.from(context).inflate(R.layout.category, viewGroup, false);
+        return new CategoryViewHolder(view);
```

Comment:
@SimonMaha You should bind click listener here

CategoryViewHolder viewHolder = new CategoryViewHolder(view);
viewHolder.someView.setOnclickListener(onItemCLickListener);